tools/windows/install-help/cal/customactiondata.cpp

```diff
@@ -266,11 +266,25 @@ void CustomActionData::ensureDomainHasCorrectFormat()
             WcaLog(LOGMSG_STANDARD, "Supplied domain name \"%S\"", _user.Domain.c_str());
             _domainUser = true;
         }
-        else if (_wcsicmp(_user.Domain.c_str(), L"NT AUTHORITY") != 0) // NT Authority should never be considered a "domain".
+        else
         {
-            WcaLog(LOGMSG_STANDARD, "Warning: Supplied user in different domain (\"%S\" != \"%S\")", _user.Domain.c_str(),
-                   _targetMachine->DnsDomainName().c_str());
-            _domainUser = true;
+            // Compute a temporary fully qualified username to retrieve its SID
+            // in order to determine if its prefix starts with NT AUTHORITY.
+            auto tempFquname = _user.Domain + L"\\" + _user.Name;
+            auto sidResult = GetSidForUser(nullptr, tempFquname.c_str());
+            const auto ntAuthoritySid = WellKnownSID::NTAuthority();
+            if (!ntAuthoritySid.has_value())
+            {
+                WcaLog(LOGMSG_STANDARD, "Cannot check user SID against NT AUTHORITY: memory allocation failed");
+            }
+            else if (!EqualPrefixSid(
+                         sidResult.Sid.get(),
```

Comment:
Could sidResult.Sid be null just like ntAuthoritySid

Reply:
You know better. I am judging from the small scope of the code itself. One var is checked but another seems to be not.